ivre/db/sql/__init__.py

```diff
@@ -1142,6 +1144,57 @@ def _migrate_schema_17_18(self):
         )
         return len(failed)
 
+    def _migrate_schema_18_19(self):
+        """Converts a record from version 18 to version 19. Version 19
+splits smb-os-discovery scripts into two, a smb-ntlm-info one that contains all
+the information the original smb-os-discovery script got from NTLM, and a
+smb-os-discovery script with only the information regarding SMB
+
+        """
+        failed = set()
+        req = (select([self.tables.scan.id,
+                       self.tables.script.name,
+                       self.tables.script.port,
+                       self.tables.script.output,
+                       self.tables.script.data])
+               .select_from(join(join(self.tables.scan, self.tables.port),
+                                 self.tables.script))
+               .where(and_(self.tables.scan.schema_version == 18,
+                           self.tables.script.name == "smb-os-discovery")))
+        for rec in self.db.execute(req):
+            if rec.name in rec.data:
+                try:
+                    smb, ntlm = xmlnmap.split_smb_os_discovery(rec.data)
+                except Exception:
+                    utils.LOGGER.warning("Cannot migrate host %r", rec.id,
+                                         exc_info=True)
+                    failed.add(rec.id)
+                else:
+                    if 'masscan' in smb:
+                        data = {'smb-os-discovery': smb['smb-os-discovery'],
+                                'masscan': smb['masscan']}
+                    else:
+                        data = smb['smb-os-discovery']
```

Comment:
This is a bit weird, or do I miss something? The data structure will be very different between data from Nmap and Masscan if I understand this well.